cool.py

```diff
@@ -545,7 +545,18 @@ def append_results(results_df):
     )
 
     job = BIGQUERY_CLIENT.load_table_from_dataframe(results_df, table_id, job_config=job_config, location="US")
-    job.result()  # Waits for table load to complete.
+   
+    try:
+        job_result = job.result()  # Waits for table load to complete.
+    except Exception as ex:
+        logging.error("unable to append rows into the results table! %s", ex)
+
+        return
+
+    if job_result.total_rows == 0:
+        logging.warning("no rows were appended into the results table!")
```

Comment:
**question**: would this be the no detections case? If so, we don't want to be warning about that.

Reply:
No, this function should only run if there are good detections to append into the results table.  Otherwise, the continue statement should proceed to the next iteration and `append_results` and `update_index` will not be called if there are no detections.